## PyPhylogenomics Outline

1. Resolved phylogenies are needed because can be applied to studies using the
        comparative method [@felsenstein1985].
    * phylogenomic approaches are needed to resolve the tree of life in diverse
      clades [@eisen2003].
 	* many clades therefore need to move beyond 5 - 10 genes, and begin
      accessing 100 to 1000's of genes due to, for example, coalescent
      stochasticity [@leache2011].
	* genes to use, etc.
2. Whole genome sequencing
    * Nowadays, whole-genome datasets of related species are being used to
      resolve species phylogenies of some taxa of interest for example mammals
      [@romiguier2013], molluscs [@kocot2013], horses [@vilstrup2013] but this is
      still costly [@metzker2010].

3. Amplicon sequencing (target sequencing)

    * a few genes: Turner, E.H., Lee, C., Ng, S.B., Nickerson, D.A. & Shendure,
    J. Massively parallel exon capture and library-free resequencing across
    16 genomes. Nat. Methods 6, 315â€“316 (2009.

    * or all exons (exome)

3. Transcriptome analysis

	* @regier2013 extract mRNAs and use reverse transcription to synthetize
      cDNA using primers for certain genes of interest (up to 19 genes). The cDNA
      is amplified and sequenced using traditional Sanger sequencers.

    * RNA sequencing (RNA-Seq), whole transcriptome RNA is used to generate cDNA
      for Next Gen Seq into EST libraries for assembly. This allows estimating
      gene structure such as start, end and splicing patterns [@wang2009]. Used
      in *Melitaea cinxia* [@vera2008].

    * most research groups are doing this [@cui2013].

	* The i5K initiative aims to sequence 5,000 arthropod genomes [@i5k2013].

	* The Genome 10K initiative aims to sequence 10,000 vertebrate genomes
      [@g10k].

4. Hybrid enrichment (sequence capture)

    * Hybrid enrichment (a.k.a. sequence capture) uses either
      synthetic oligonucleotides as baits that hybridize to genomic fragments
      containing the target sequence and allows target loci [@albert2007;
      @gnirke2009].

    * PCR products from the mitochondrial genome have been
      used as probes for hybridization and sequencing [@maricic2010].

    * Exons for one species identified via the transcriptome analysis have been
      used for capturing the exome of related taxa [e.g. in chipmunks @bi2012].

    * "Pull-down" method by @lemmon2012 use a few model species to design
      highly conserved probes of short genomic sequences flanked by less
      conserved regions, which are captured using hybridization with genomes of
      non-model species.

5. RADseq, 

4. Problems with the transcriptome approach 
	* not going to work on degraded museum specimens. 
	* what is needed .... long exons for single copy genes
	* Wahlberg and Wheat were focused upon generating markers compatible with 
      such material. 
	* Here we extend that approach with a tool for the community

5. Targeted re-sequencing,

## References
